arangod/Agency/RestAgencyPrivHandler.cpp

```diff
@@ -246,8 +246,9 @@ RestStatus RestAgencyPrivHandler::execute() {
             redirectRequest(slice.get("id").copyString());
             return RestStatus::DONE;
           }
-          for (auto const& obj : VPackObjectIterator(ret->slice())) {
-            result.add(obj.key.copyString(), obj.value);
+          for (auto obj : VPackObjectIterator(
```

Comment:
having `auto` instead of `auto const&` now makes a copy. However, the type returned by `operator*` is just two pointers, so its cheap to copy. If the implementation would not be visible to the translation unit, this would have been translated to a pass by reference return value, since the ABI only allows for a single register as return value. (and making it `auto` instead of `auto const&` would have had no effect)

But the implementation is known, so it will be optimized. In both cases, the compiler produces identical code. https://godbolt.org/z/fPo1oa1P8